### PR TITLE
Bring the tests for the interface module

### PR DESCRIPTION
**Pull request by rodis**
DHC-1048
We copied the interface module from quantum into the akanda-rug
without its test. This fix brings in the tests for the module
and adapt them to run without the quantun as dependency.

Signed-off-by: Rosario Di Somma rosario.disomma@dreamhost.com
Change-Id: I7136c9bd367532e12e8c58fb4469f100d10f4ed9

### CODE REVIEW

File: test/unit/common/test_linux_interface.py [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+            self.device_exists_p.stop()
+        except RuntimeError:
+            pass
+        self.ip_dev_p.stop()
+        self.ip_p.stop()
+        # Note(rods): some tests override the default options so let's reset
+        # the config object
+        self.conf.reset()
+
+class TestABCDriver(TestBase):
```
Comment (dhellmann) [inline test/unit/common/test_linux_interface.py:85]: pep8 didn't complain about this line?

Comment (markmcclain) [inline test/unit/common/test_linux_interface.py:85]: Travis isn't running on this project.

**Comment by dhellmann** [comment]
LGTM, I'll let Mark approve since he's more familiar with the tests.

**Comment by markmcclain** [comment]
This only partially covers the tests we need to import.  We also need to bring in tests for ip_lib.py, ovs_lib.py, and utils.py.  Should we file separate tickets for these?

**Comment by dhellmann** [comment]
Let's get them all in one changeset, if we can.

**Comment by rodis** [comment]
Ok, I'll bring in the tests for the other modules

**Comment by rodis** [comment]
I added the tests for the ip_lib.py, ovs_lib.py, and utils.py modules, this should cover all the tests we need, let me know if I missed something.

**Comment by dhellmann** [comment]
LGTM

**Pull request merged**